generator/services_test.go

```diff
@@ -138,6 +138,52 @@ func TestServices_ManualIngressNetwork(t *testing.T) {
 	}, expectedCaddyfile, expectedLogs)
 }
 
+func TestServices_OverrideIngressNetwork(t *testing.T) {
+	dockerClient := createBasicDockerClientMock()
+	dockerClient.NetworksData = []types.NetworkResource{
+		{
+			ID:   "other-network-id",
+			Name: "other-network-name",
+		},
+		{
+			ID:   "another-network-id",
+			Name: "another-network-name",
+		},
+	}
+	dockerClient.ServicesData = []swarm.Service{
+		{
+			ID: "SERVICE-ID",
+			Spec: swarm.ServiceSpec{
+				Annotations: swarm.Annotations{
+					Name: "service",
+					Labels: map[string]string{
+						"caddy_ingress_network":      "another-network",
+						fmtLabel("%s"):               "service.testdomain.com",
+						fmtLabel("%s.reverse_proxy"): "{{upstreams}}",
+					},
+				},
+			},
+			Endpoint: swarm.Endpoint{
+				VirtualIPs: []swarm.EndpointVirtualIP{
+					{
+						NetworkID: "other-network-id",
+					},
+				},
+			},
+		},
+	}
+
+	const expectedCaddyfile = "service.testdomain.com {\n" +
+		"	reverse_proxy service\n" +
+		"}\n"
+
+	const expectedLogs = otherIngressNetworksMapLog + swarmIsAvailableLog
+
+	testGeneration(t, dockerClient, func(options *config.Options) {
+		options.IngressNetworks = []string{"other-network-name"}
```

Comment:
You're right, this test is not meaningful as it is right now.
But swarm service proxying supports 2 modes, using dns name, or proxying to ips.
Add another option after this line like:
```
options.ProxyServiceTasks = true
```
This will swap the behaviour, and then adjust the `expectedCaddyfile` to match the IP addresses configured in `dockerClient.ServicesData`

Reply:
Ah, I see. If I understand correctly, I should use `TestServiceTasks_ManualIngressNetwork` as a reference. Is overriding the ingress network to a virtual service something that should be tested?

Reply:
No need